### DISCUSSION TITLE
Why the metrics data will be reset?

### DISCUSSION THREAD

**Discussion by zby-universal** [prometheus/client_python#941] [Q&A]
I have some metrics and the label may have many values. 
![image](https://github.com/prometheus/client_python/assets/60498735/38dd68a2-db48-4acb-8d1f-95d267b356de)
So will the number of the metrics exceed some threshold and all metrics wiil be reset? What is the maximum threshold for the number of monitoring indicators？

**Comment by csmarchbanks** [comment]
Counter resets typically happen when a service restarts, and are handled by functions such as `rate` in Prometheus.

**Comment by zby-universal** [comment]
Actually, I have checked all machines which did't restart and the service just used counter/histogram metric to record data. But from Prometheus service side, the counter metrics data of all machines appear negative value like as follow: 
PS: The service is deployed with 20 instance and multi-proccess way,  and I use multi-process metrics method to record the data.

![image](https://github.com/prometheus/client_python/assets/60498735/7317707a-a90b-4e14-bbee-b208a8b28330)
